internal/infrastructure/kubernetes/infra_resource.go

```diff
@@ -119,6 +119,16 @@ func (i *Infra) createOrUpdateDeployment(ctx context.Context, r ResourceRender)
 	}
 
 	defer func() {
+		// custom Deployment name is enabled, and the deployment name is different from the resource name.
```

Comment:
this takes into account when default name changes to custom name, what about the case when custom name is removed and we switch to default name ?
my suggestion would be instead do a `Client.List` and delete all that are not current

Reply:
we should use the one which not the currnet one.

Reply:
good catch!